taskscheduler.ts

```diff
@@ -1,15 +1,11 @@
 const resolved = Promise.resolve(undefined);
 
 // Returns a function that asynchronously schedules a task
-export function createTaskScheduler() : TaskScheduler {
+export function createTaskScheduler(scheduleMacrotask: TaskScheduler) : TaskScheduler {
```

Comment:
Make it optional argument, defaulting to `tiny-set-immediate`? Then the code from `asynciterator.ts` can stay as-is; it shouldn't need to know about internals.